Changing some IllegalArgumentException to MirrorException

garcia-jj opened the pull request:
Changing some `IllegalArgumentException` to `MirrorException` because validates logic and not arguments. `IllegalArgumentException` are designed to check arguments, and not results from some logic process like a when a method isn't found. Some places already uses `MirrorException`, so I changed to this exception.

garcia-jj committed:
Changing some IllegalArgumentException to MirrorException because
validates logic, not arguments

jonasabreu commented:
That does make sense. The rationale behind the use of IAE instead of ME was that we expected the user to pass an existing field name. But now this looks more like a logic validation (since this is already too related to reflection).

Did you only take a look at these places or did you thoroughly check the code base? Just to know if there is need to review the rest of the code after merging this.

garcia-jj committed:
Adding missing test class

garcia-jj commented:
I added a missing test class that I changed. I checked all code to see if there are more places with this exceptions, but I didn't found.

There are a 3 unit test failures, but not related to this  pull request because already fails into master. I can try to fix in another pull request, if you want.

Thank you.

jonasabreu commented:
No need to look at the failing tests. I'll check them out later. Thanks for the PR!
